client/my-sites/comments/comment/comment-html-editor/add-image-dialog.jsx

```diff
@@ -0,0 +1,93 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+
+export class AddImageDialog extends Component {
+	static propTypes = {
+		onClose: PropTypes.func,
+		onInsert: PropTypes.func,
+		shouldDisplay: PropTypes.bool,
+		translate: PropTypes.func,
+	};
+
+	state = {
+		imageAlt: '',
+		imageTitle: '',
+		imageUrl: '',
+	};
+
+	setImageAlt = event => this.setState( { imageAlt: event.target.value } );
+
+	setImageTitle = event => this.setState( { imageTitle: event.target.value } );
+
+	setImageUrl = event => this.setState( { imageUrl: event.target.value } );
+
+	closeDialog = () => {
+		this.setState( {
+			imageAlt: '',
+			imageTitle: '',
+			imageUrl: '',
+		} );
+		this.props.onClose();
```

Comment:
it's probably good to pass `this.props.onClose` as the second argument to `this.setState()` because otherwise we're exposing a potential race condition between calling that function and updating state. if we send it as an argument it will only be called after `setState()` has actually updated local state.

```js
this.setState( { imageAlt: '', imageTitle: '', imageUrl: '' }, this.props.onClose );
```

Note that we're not _calling_ `onClose` but rather _passing_ it as a callback.